cmd/compute-domain-kubelet-plugin/computedomain.go

```diff
@@ -253,6 +255,66 @@ func (m *ComputeDomainManager) AssertComputeDomainReady(ctx context.Context, cdU
 	return nil
 }
 
+func (m *ComputeDomainManager) AddComputeDomainNodeStatus(ctx context.Context, cdUID string) error {
+	cd, err := m.GetComputeDomain(ctx, cdUID)
+	if err != nil {
+		return fmt.Errorf("error getting ComputeDomain: %w", err)
+	}
+	if cd == nil {
+		return fmt.Errorf("ComputeDomain not found: %s", cdUID)
+	}
+
+	if cd.Status.Status == nvapi.ComputeDomainStatusReady {
+		return nil
+	}
+
+	var nodeNames []string
+	for _, node := range cd.Status.Nodes {
+		nodeNames = append(nodeNames, node.Name)
+	}
+
+	if slices.Contains(nodeNames, m.config.flags.nodeName) {
+		return nil
+	}
+
+	node, err := m.GetComputeDomainNodeStatusInfo(ctx, m.config.flags.nodeName)
+	if err != nil {
+		return fmt.Errorf("error getting ComputeDomain node status info: %w", err)
+	}
+
+	newCD := cd.DeepCopy()
+	newCD.Status.Nodes = append(newCD.Status.Nodes, node)
+	newCD.Status.Status = nvapi.ComputeDomainStatusNotReady
+	if _, err = m.config.clientsets.Nvidia.ResourceV1beta1().ComputeDomains(newCD.Namespace).UpdateStatus(ctx, newCD, metav1.UpdateOptions{}); err != nil {
+		return fmt.Errorf("error updating nodes in ComputeDomain status: %w", err)
+	}
+
+	return nil
+}
+
+func (m *ComputeDomainManager) GetComputeDomainNodeStatusInfo(ctx context.Context, nodeName string) (*nvapi.ComputeDomainNode, error) {
+	node, err := m.config.clientsets.Core.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error getting Node '%s': %w", nodeName, err)
+	}
+
+	var ipAddress string
+	for _, addr := range node.Status.Addresses {
+		if addr.Type == corev1.NodeInternalIP {
+			ipAddress = addr.Address
+			break
+		}
+	}
+
+	n := &nvapi.ComputeDomainNode{
+		Name:      nodeName,
+		IPAddress: ipAddress,
+		CliqueID:  m.cliqueID,
```

Comment:
Note to self: we set `m.cliqueID` (string) upon calling `NewComputeDomainManager()`. That is probably populated straight from source of truth (code not shown in diff).

Reply:
yes, this was already available for other reasons